geomesa-fs/geomesa-fs-datastore/src/main/scala/org/locationtech/geomesa/fs/FileSystemFeatureIterator.scala

```diff
@@ -0,0 +1,151 @@
+/***********************************************************************
+ * Copyright (c) 2013-2017 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+
+package org.locationtech.geomesa.fs
+
+import java.util
+import java.util.concurrent._
+
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.hadoop.fs.FileSystem
+import org.geotools.data.Query
+import org.locationtech.geomesa.fs.storage.api.{FileSystemStorage, Partition, PartitionScheme}
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+
+import scala.util.control.NonFatal
+
+class FileSystemFeatureIterator(fs: FileSystem,
+                                partitionScheme: PartitionScheme,
+                                sft: SimpleFeatureType,
+                                q: Query,
+                                readThreads: Int,
+                                storage: FileSystemStorage) extends java.util.Iterator[SimpleFeature] with AutoCloseable {
+
+  private val partitions = FsQueryPlanning.getPartitionsForQuery(storage, sft, q)
+
+  private val iter: java.util.Iterator[SimpleFeature] with AutoCloseable =
+    if (partitions.isEmpty) {
+      CloseableEmptyIterator
+    } else {
+      new ThreadedReader(storage, partitions, sft.getTypeName, q, readThreads)
+    }
+
+  override def hasNext: Boolean = iter.hasNext
+  override def next(): SimpleFeature = iter.next()
+  override def close(): Unit = iter.close()
+}
+
+object CloseableEmptyIterator extends java.util.Iterator[SimpleFeature] with AutoCloseable  {
```

Comment:
Can you call org.locationtech.geomesa.utils.collection.CloseableIterator.empty[SimpleFeature] instead of making this object?